docs/src/test/scala/docs/http/scaladsl/server/directives/CustomDirectivesExamplesSpec.scala

```diff
@@ -77,4 +79,31 @@ class CustomDirectivesExamplesSpec extends RoutingSpec {
     //#flatMap-0
   }
 
+  "scratch" in {
+    //#scratch
+    object HostnameAndPort extends Directive[(String, Int)] {
+
+      private val hostnameAndPort =
+        textract(ctx => {
+          val authority = ctx.request.uri.authority
+          (authority.host.address(), authority.port)
+        })
+
+      override def tapply(f: ((String, Int)) => Route): Route = hostnameAndPort {
```

Comment:
Thanks a lot for your reviews @jrudolph!

I think it's better to provide both to give a more comprehensive guide. But when I run the example above, I get an error below:
```scala
Error:(13, 6) type mismatch;
 found   : akka.http.scaladsl.server.Route
    (which expands to)  akka.http.scaladsl.server.RequestContext => scala.concurrent.Future[akka.http.scaladsl.server.RouteResult]
 required: scala.concurrent.Future[akka.http.scaladsl.server.RouteResult]
    f((authority.host.address(), authority.port))
```
Following the error message, I add an explicit parameter to make it works:
```scala
object hostnameAndPort extends Directive[(String, Int)] {
  override def tapply(f: ((String, Int)) ⇒ Route): Route = { ctx => // remember: Route = RequestContext => Future[RouteResult]
    val authority = ctx.request.uri.authority
    f((authority.host.address(), authority.port))(ctx)
  }
}

def hostnameAndPort: Directive[(String, Int)] = Directive[(String, Int)] { inner => ctx =>
  val authority = ctx.request.uri.authority
  inner((authority.host.address(), authority.port))(ctx)
}
```
This is a little awkward, and any suggestion to avoid the implicit `ctx` parameter? Thanks :)

Reply:
Ah, sorry, I missed that. Yes, you need the `ctx` parameter.

Reply:
Get it.